File: Python/프로그래머스/Level2/문자열 압축.py
s = "aabbaccc"


def solution(s):
    answer = 0
    candi = []
    for cutlen in range(1, len(s)//2+1):
        i = 0
        cnt = 0
        pattern = ''
        news = ''
        while (1):
            if i >= len(s):
                break
            cur = s[i:i+cutlen]
            next = s[i+cutlen:i+cutlen+cutlen]
            if cur == next and pattern == cur:
                cnt += 1
                i += cutlen
            elif cur == next and pattern != cur:
                news += str(cnt)+pattern
                pattern = cur
                cnt = 2
                i += cutlen
            else:
                news += cur

        print(news)
    return answer


solution(s)

File: Javascript/프로그래머스/Level2/문자열 압축.js
let s = 'aabbaccc';

function solution(s) {
    var answer = 0;
    let half = parseInt(s.length / 2);

    for (let i = 1; i <= half; i++) {
        // 문자를 자르는 단위
        let cur = 0; // 체크위치
        let newString = '';
        while (1) {
            if (cur === s.length - i) break;
            let lcnt = 1;
            if (s.substring(cur, cur + i + 1) === s.substring(cur + i, cur + i + i + 1)) {
                cur += i;
                lcnt += 1;
                console.log('cur: ' + s.subString(cur, cur + i));
                console.log(s.subString(cur + i, cur + i + i));
            }
        }
    }

    return answer;
}

solution(s);
